coderd/aitasks.go

```diff
@@ -590,3 +597,209 @@ func (api *API) taskDelete(rw http.ResponseWriter, r *http.Request) {
+	if err != nil {
+		return httperror.NewResponseError(http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error fetching workspace resources.",
+			Detail:  err.Error(),
+		})
+	}
+	if len(data.builds) == 0 || len(data.templates) == 0 {
+		return httperror.ErrResourceNotFound
+	}
+	build := data.builds[0]
+	if build.HasAITask == nil || !*build.HasAITask || build.AITaskSidebarAppID == nil || *build.AITaskSidebarAppID == uuid.Nil {
+		return httperror.NewResponseError(http.StatusBadRequest, codersdk.Response{
+			Message: "Task is not configured with a sidebar app.",
+		})
+	}
+	sidebarAppID := *build.AITaskSidebarAppID
+
+	// Find the sidebar app details to get the URL and validate app health.
+	agentIDs := make([]uuid.UUID, 0, len(build.Resources))
+	for _, res := range build.Resources {
+		for _, agent := range res.Agents {
+			agentIDs = append(agentIDs, agent.ID)
+		}
+	}
+	// TODO(mafredri): Can we avoid dbauthz.AsSystemRestricted(ctx)?
+	//nolint:gocritic // GetWorkspaceAppsByAgentIDs requires system restricted context.
+	apps, err := api.Database.GetWorkspaceAppsByAgentIDs(dbauthz.AsSystemRestricted(ctx), agentIDs)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return httperror.NewResponseError(http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error fetching workspace apps.",
+			Detail:  err.Error(),
+		})
+	}
+	sidebarApp, ok := slice.Find(apps, func(app database.WorkspaceApp) bool { return app.ID == sidebarAppID })
+	if !ok {
+		return httperror.NewResponseError(http.StatusBadRequest, codersdk.Response{
+			Message: "Task sidebar app not found in latest build.",
+		})
+	}
+
+	// Return an informative error if the app isn't healthy rather than trying
+	// and failing.
+	switch sidebarApp.Health {
+	case database.WorkspaceAppHealthDisabled:
+		// No health check, pass through.
+	case database.WorkspaceAppHealthInitializing:
+		return httperror.NewResponseError(http.StatusServiceUnavailable, codersdk.Response{
+			Message: "Task sidebar app is initializing. Try again shortly.",
+		})
+	case database.WorkspaceAppHealthUnhealthy:
+		return httperror.NewResponseError(http.StatusServiceUnavailable, codersdk.Response{
+			Message: "Task sidebar app is unhealthy.",
+		})
+	}
+
+	// Build the direct app URL and dial the agent.
+	if !sidebarApp.Url.Valid || sidebarApp.Url.String == "" {
+		return httperror.NewResponseError(http.StatusInternalServerError, codersdk.Response{
+			Message: "Task sidebar app URL is not configured.",
+		})
+	}
+	parsedURL, err := url.Parse(sidebarApp.Url.String)
+	if err != nil {
+		return httperror.NewResponseError(http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error parsing task app URL.",
+			Detail:  err.Error(),
+		})
+	}
+	if parsedURL.Scheme != "http" {
+		return httperror.NewResponseError(http.StatusBadRequest, codersdk.Response{
+			Message: "Only http scheme is supported for direct agent-dial.",
+		})
+	}
```

Comment:
Do we even support having SSL in workspace apps right now?

Reply:
I'm honestly not sure, but better safe than sorry? We could support it easily here by setting the http client to ignore certificates but better not unless there's a need.